src/Grpc/GrpcUnaryCall.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/*
+ * Copyright 2016, Google Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+namespace Google\GAX\Grpc;
+
+use Google\GAX\ApiException;
+use Google\GAX\UnaryCall;
+use InvalidArgumentException;
+
+class GrpcUnaryCall implements UnaryCall
+{
+    private $innerUnaryCall;
+
+    public function __construct(\Grpc\UnaryCall $innerUnaryCall)
+    {
+        $this->innerUnaryCall = $innerUnaryCall;
+    }
+
+    public function wait()
```

Comment:
Correct me if I'm wrong, but since [OperationsGapicClient](https://github.com/googleapis/gax-php/blob/e09f219ed50ea1f6162d1945156d2ea6af6cd1f9/src/LongRunning/Gapic/OperationsGapicClient.php#L311) expects a callable, if we want to keep the interface, then in `OperationsGapicClient::getOperation`, will need to call `$wait`:

```php
    public function getOperation($name, $optionalArgs = [])
    {
        $request = new GetOperationRequest();
        $request->setName($name);

        $mergedSettings = $this->defaultCallSettings['getOperation']->merge(
            new CallSettings($optionalArgs)
        );
        $unaryCall = ApiCallable::createApiCall(
            $this->operationsStub,
            'GetOperation',
            $mergedSettings,
            $this->descriptors['getOperation']
        );
        return $unaryCall->wait();
    }
```

We could make the unary call into a callable (i.e. implement `__invoke`), but then we lose the ability to do async later on.

Reply:
`piCallable::createApiCall` currently returns a `callable` - I was imagining it would still return a `callable`. This could either (a) be a callable that is used exactly as it is now, or (b) be a callable that returns a `UnaryCall` object, in which case yes we would need to call wait.

I think I prefer (a), but not 100% sure

Reply:
I agree, I kept it a callable and then return wait on the callable in the new PR.

Reply:
There is an interesting side-effect to this, which is that now the retry logic does not work, as an exception is not thrown until _after_ wait is called.

Reply:
This is a great point. This seems problematic in general for any sort of scheme to combine async calls and retries, because we have no access to the result of the initial call without calling wait() on the grpc return object.

Perhaps what is required is that the <API>TransportInterface methods still returns a GAX\UnaryCall interface object, but that we call wait inside the apicallable, much like existing code here: https://github.com/googleapis/gax-php/blob/master/src/ApiCallable.php#L143 (although the logic to convert from a status to an exception will already be handled.)

That would mean we wouldn't need to call wait() in the method implementation in the Gapic client, and then retries should function as expected.

Reply:
It seems to me that we may be able to have a "call" object which owns a call stack, and the call stack can either be called synchronously or asynchronously. 

```php
$call = $transport->createCall($method);
$call->setCallStack(ApiCallable::createCallStack($options));
```